lambeq/text2diagram/depccg_parser.py

```diff
@@ -119,9 +117,38 @@ def __init__(self,
                              '\'progress\' level of verbosity. '
                              f'`{self.verbose}` was given.')
         _import_depccg()
+        if lang == 'en':
+            self.tokenize = False
+            if root_cats is None:
+                root_cats = [
+                    'S[dcl]', 'S[wq]', 'S[q]', 'S[qem]', 'NP']
+            self.annotator_fun = english_annotator.get(annotator, annotate_XX)
+        elif lang == 'ja':
+            if annotator is None:
+                annotator = 'janome'
+            self.tokenize = True
```

Comment:
So tokenization in Japanese should be always on? Or this is just the preferred default? Because in the first case, we need to also add safe-nets in the individual methods.

Reply:
Yes. Japanese is not a sentence in which words are separated by spaces. Therefore, tokenize is true. Janome is a tokenizer that performs morphological analysis.

Reply:
While this makes sense, we need to keep an option to not tokenise to allow the user to pass in pre-tokenised sentences. They can always specify `DepCCGParser(..., annotator=joname)` if required.

Reply:
@KentaroAOKI Ok, based on the links you provide, it seems you cannot use depccg with `lang=ja` and pre-tokenised inputs. In that case I would suggest the following:

1. Remove the global `self.tokenise` attribute you have added to depccg parser. This is only used in `_depccg_parse()`, where you can use a condition anyway (or perhaps use a `partial`ly initialised function, from `functools` package)
2. Keep as default the Japanese annotator you consider more common, as you already did.
3. In `sentence2tree()`and `sentence2trees()`, add a guard that does not allow pre-tokenised sentences when language is Japanese. Note there is already some code that checks the inputs and validates them against the passed argument `tokenised`, this would be the right place for you to add the new conditions.

Something else: Can you please check the box "Allow edits from maintainers" in your repo so we could also help with the PR? For example, before merging we would need to update the docs and add a tutorial page for using DepCCG with Japanese support, based on your PR description.

This is really useful work, thanks for taking the time to put this together!

Reply:
@dimkart Thanks, The box "Allow edits from maintainers" is selected.

Reply:
Sorry for the late reply.

depccg appears to show an example of using it with pre-tokenised inputs:
```
➜ echo "これ は テスト の 文 です 。" | depccg_ja --pre-tokenized
ID=1, Prob=-53.98793411254883
{< S[mod=nm,form=base,fin=t] {< S[mod=nm,form=base,fin=f] {< NP[case=nc,mod=nm,fin=f] {NP[case=nc,mod=nm,fin=f] これ/これ/**} {NP[case=nc,mod=nm,fin=f]\NP[case=nc,mod=nm,fin=f] は/は/**}} {< S[mod=nm,form=base,fin=f]\NP[case=nc,mod=nm,fin=f] {< NP[case=nc,mod=nm,fin=f] {< NP[case=nc,mod=nm,fin=f] {NP[case=nc,mod=nm,fin=f] テスト/テスト/**} {NP[case=nc,mod=nm,fin=f]\NP[case=nc,mod=nm,fin=f] の/の/**}} {NP[case=nc,mod=nm,fin=f]\NP[case=nc,mod=nm,fin=f] 文/文/**}} {(S[mod=nm,form=base,fin=f]\NP[case=nc,mod=nm,fin=f])\NP[case=nc,mod=nm,fin=f] です/です/**}}} {S[mod=nm,form=base,fin=t]\S[mod=nm,form=base,fin=f] 。/。/**}}
```

and from https://github.com/masashi-y/depccg/blob/master/depccg/__main__.py#L42 it seems like it does have a default annotator when not tokenising. Does this work for you, or am I mistaken?

Reply:
Hi @ianyfan, Thank you for your reply. I checked it. You are right. Can I add tokenize arguments to __init__ to determine the annotator?

Reply:
Is the `annotator` argument not sufficient?

Reply:
Sorry for the late reply. I will check over the weekend.

Reply:
Thank you. I checked the modified code.